tests/std/tests/P2286R8_text_formatting_range_set/test.cpp

```diff
@@ -0,0 +1,1773 @@
+    check_exception("Type ?s requires character type as formatting argument", SV("{:?s}"), input);
+
+    for (basic_string_view<CharT> fmt : fmt_invalid_types<CharT>("s")) {
+        check_exception("The format specifier should consume the input or end with a '}'", fmt, input);
+    }
+
+    // ***** Only underlying has a format-spec
+    check(SV("{  0x0}"), SV("{::5}"), input);
+    check(SV("{0x0**}"), SV("{::*<5}"), input);
+    check(SV("{_0x0_}"), SV("{::_^5}"), input);
+    check(SV("{::0x0}"), SV("{:::>5}"), input);
+
+    check(SV("{  0x0}"), SV("{::{}}"), input, 5);
+    check(SV("{0x0**}"), SV("{::*<{}}"), input, 5);
+    check(SV("{_0x0_}"), SV("{::_^{}}"), input, 5);
+    check(SV("{::0x0}"), SV("{:::>{}}"), input, 5);
+
+    check_exception("The format string contains an invalid escape sequence", SV("{::}<}"), input);
+    check_exception("The fill option contains an invalid value", SV("{::{<}"), input);
+
+    // *** sign ***
+    check_exception("The format specifier should consume the input or end with a '}'", SV("{::-}"), input);
+
+    // *** alternate form ***
+    check_exception("The format specifier should consume the input or end with a '}'", SV("{::#}"), input);
+
+    // *** zero-padding ***
+    check(SV("{0x0000}"), SV("{::06}"), input);
+    check(SV("{0x0000}"), SV("{::06p}"), input);
+    check(SV("{0X0000}"), SV("{::06P}"), input);
+
+    // *** precision ***
+    check_exception("The format specifier should consume the input or end with a '}'", SV("{::.}"), input);
+
+    // *** locale-specific form ***
+    check_exception("The format specifier should consume the input or end with a '}'", SV("{::L}"), input);
+
+    // *** type ***
+    for (basic_string_view<CharT> fmt : fmt_invalid_nested_types<CharT>("pP")) {
+        check_exception("The type option contains an invalid value for a pointer formatting argument", fmt, input);
+    }
+
+    // ***** Both have a format-spec
+    check(SV("^^{::0x0}^^^"), SV("{:^^12::>5}"), input);
+    check(SV("^^{::0x0}^^^"), SV("{:^^{}::>5}"), input, 12);
+    check(SV("^^{::0x0}^^^"), SV("{:^^{}::>{}}"), input, 12, 5);
+
+    check(SV("^^{::0x0}^^^"), SV("{:^^12::>5}"), input);
+    check(SV("^^{::0x0}^^^"), SV("{:^^{}::>5}"), input, 12);
+    check(SV("^^{::0x0}^^^"), SV("{:^^{}::>{}}"), input, 12, 5);
+
+    check_exception(
+        "The argument index value is too large for the number of arguments supplied", SV("{:^^{}::>5}"), input);
+    check_exception(
+        "The argument index value is too large for the number of arguments supplied", SV("{:^^{}::>{}}"), input, 12);
+}
+
+template <class CharT, class TestFunction, class ExceptionTest>
+void test_pointer(TestFunction check, ExceptionTest check_exception) {
+    // Note nullptr_t can only be equality compared so not used in a set.
+    test_pointer<CharT>(check, check_exception, unordered_set{static_cast<const void*>(0)});
+    test_pointer<CharT>(check, check_exception, unordered_multiset{static_cast<void*>(0)});

```

Comment:
Starting from `0` is too gross for me to accept in our codebase, changing this to `nullptr` (which doesn't affect the test behavior).